### DISCUSSION TITLE
error after running `prisma introspect` - Decimal vs Float

### DISCUSSION THREAD

**Discussion by alan345** [prisma/prisma#6421] [Q&A] [answered]
My MySQL database is for the field productCost: `decimal(65,30)`
My `shema.prisma` was:
```prisma
productCost                       Float
```

I ran `prisma introspect` now, my `shema.prisma` is:
```prisma
productCost                       Decimal
```
It now drops an error in the code as soon as I do an Mathematic operation: 
```The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.```

![image](https://user-images.githubusercontent.com/15246526/113630097-348f7680-961c-11eb-9740-79c699688580.png)

What is the recommendation?

**Comment by alan345** [comment]
Related topic: https://github.com/prisma/prisma/issues/6049#issuecomment-805655549

**Reply by alan345** [reply]
Adding some conversation about this: 
![image](https://user-images.githubusercontent.com/15246526/114057966-e6f75180-9847-11eb-8131-6e7282520144.png)

**Reply by ryands17** [reply]
@alan345 for reference: https://github.com/MikeMcl/decimal.js/
Mathematic operations can be done via the methods it provides, so something like this: 

```ts
invoice.productCost.mul(2)
```

**Reply by alan345** [reply]
Hi @ryands17 
This is my point. I dont want to change my entire codebase and use this library https://github.com/MikeMcl/decimal.js/

As you said here: https://github.com/prisma/prisma/issues/7158 libraries change over time. I dont want to be tied to it.

I need to use the native Math JS operators. 
```invoice.productCost * 2 `

Save in Decimals in the database use as float in the code.

**Reply by ryands17** [reply, marked as answer]
I don't think that's possible at the moment. You would need to follow the above request but `Decimal.js` was added for a reason that precision was required for floats that JS doesn't provide.

You can read more about the comment posted here and the reason for this, so it's unlikely that `Decimal.js` will be removed in the near future: https://github.com/prisma/prisma/issues/6049#issuecomment-851461382

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
